services/src/datasets/external/netcdfcf/database.rs

```diff
@@ -0,0 +1,1438 @@
+                &provider_id,
+                &file_name,
+                &i64::from(offset),
+                &i64::from(limit),
+            ],
+        )
+        .await
+        .boxed_context(error::UnexpectedExecution)?
+        .into_iter()
+        .map(|row| NetCdfEntity {
+            name: row.get("name"),
+            id: row.get::<_, i64>("id") as usize,
+        });
+
+    Ok(entities)
+}
+
+async fn group_metadata(
+    transaction: &Transaction<'_>,
+    provider_id: DataProviderId,
+    file_name: &str,
+    group_path: &[String],
+) -> Result<Option<NetCdfGroupMetadata>> {
+    let Some(row) = transaction
+        .query_opt(
+            r#"
+                SELECT
+                    name,
+                    title,
+                    description,
+                    data_type :: "RasterDataType",
+                    data_range,
+                    unit
+                FROM
+                    ebv_provider_groups
+                WHERE
+                    provider_id = $1 AND
+                    file_name = $2 AND
+                    name = $3
+            "#,
+            &[&provider_id, &file_name, &group_path],
+        )
+        .await
+        .boxed_context(error::UnexpectedExecution)?
+    else {
+        return Ok(None);
+    };
+
+    let metadata = NetCdfGroupMetadata {
+        name: row.get::<_, Vec<String>>("name").pop().unwrap_or_default(),
+        title: row.get("title"),
+        description: row.get("description"),
+        data_type: row.get("data_type"),
+        data_range: row.get::<_, Option<DataRange>>("data_range"),
+        unit: row.get("unit"),
+    };
+
+    Ok(Some(metadata))
+}
+
+async fn subgroup_metadata(
+    transaction: &Transaction<'_>,
+    provider_id: DataProviderId,
+    file_name: &str,
+    group_path: &[String],
+    offset: u32,
+    limit: u32,
+) -> Result<Vec<NetCdfGroupMetadata>> {
+    let subgroups = transaction
+        .query(
+            r#"
+                SELECT
+                    name,
+                    title,
+                    description,
+                    data_type :: "RasterDataType",
+                    data_range,
+                    unit
+                FROM ebv_provider_groups
+                WHERE
+                    provider_id = $1 AND
+                    file_name = $2 AND
+                    name[:$4] = $3 AND
+                    array_length(name, 1) = ($4 + 1)
+                ORDER BY name ASC
+                OFFSET $5 LIMIT $6
+                "#,
+            &[
+                &provider_id,
+                &file_name,
+                &group_path,
+                &(group_path.len() as i32),
+                &i64::from(offset),
+                &i64::from(limit),
+            ],
+        )
+        .await
+        .boxed_context(error::UnexpectedExecution)?
+        .into_iter()
+        .map(|row| NetCdfGroupMetadata {
+            name: row.get::<_, Vec<String>>("name").pop().unwrap_or_default(),
```

Comment:
isn't the name used in the group_path? Can it be empty then?

Reply:
This can be empty for the highest level.